### PR TITLE
feat(serviceID): disable id field on service edit

### PR DESCRIPTION
**Pull request by Poltergeist**
Disable the id input field in the edit form to prevent the user to change a exisiting id.

Closes DCOS-20956

### CODE REVIEW

**Comment by DanielMSchmidt** [comment]
Alfred test this please :)

**Comment by DanielMSchmidt** [comment]
😲 Alfred test this please tests all checks again? 😲

File: plugins/services/src/js/components/forms/GeneralServiceFormSection.js [hunk 1]
```diff
@@ -351,7 +351,12 @@ class GeneralServiceFormSection extends Component {
               </FormGroupHeading>
             </FieldLabel>
             <FieldAutofocus>
-              <FieldInput name="id" type="text" value={data.id} />
+              <FieldInput
+                name="id"
+                type="text"
+                value={data.id}
+                disabled={/\/edit\//g.test(global.location.hash)}
```
Comment (DanielMSchmidt) [inline plugins/services/src/js/components/forms/GeneralServiceFormSection.js]: I know this might be very nitty picky, but would you mind moving this in a variable that gives a bit more context? Something like 
```js 
const isEditPage = /\/edit\//g.test(global.location.hash);
```
maybe?

/cc @juliangieseke I know you have strong opinions against too many variables, would you still agree on this one or am I a prisoner of my own customs?

**Comment by DanielMSchmidt** [comment]
Please dismiss my review if you don't feel the same way 👍

**Comment by DanielMSchmidt** [comment]
Also @Poltergeist I tested this and I still can change the value in the JSON editor, is this a wanted behavior / out of scope for this ticket?

File: plugins/services/src/js/components/forms/GeneralServiceFormSection.js (hunk 1 above)
Comment (juliangieseke) [reply plugins/services/src/js/components/forms/GeneralServiceFormSection.js]: Well, putting this into a variable because it is hard to follow the inline regex would be a fix for a problem which should be fixed in a different way - IMHO a simple `global.location.hash.includes('edit')` would be sufficient and easier to read/understand 🤔 (assuming babel will polyfill it)

**Comment by Poltergeist** [comment]
The JSON editor should always let the user do what they want it is considered to be the advanced mode for user which do not need the form.

**Comment by Poltergeist** [comment]
Alfred you can't test this. 👯 👯‍♂️ 🎉

**Comment by DanielMSchmidt** [comment]
run integration tests

**Comment by DanielMSchmidt** [comment]
run integration tests

**Comment by Poltergeist** [comment]
run Integration Tests

**Comment by DanielMSchmidt** [comment]
@Poltergeist does this have a forward port? Just wanting to make sure we don't miss a spot
